### ISSUE TITLE
QR Code icon

### ISSUE THREAD

**Issue by Aviortheking** [feathericons/feather#598]
## Icon Request

* Icon name: qr-code
* Use case: show a qr3code on click
* Screenshots of similar icons:
font-awesome: https://fontawesome.com/icons/qrcode
mdi: https://materialdesignicons.com/icon/qrcode-scan

**Comment by johnletey** [comment]
@colebemis @locness3 @Aviortheking How does [this](https://www.figma.com/file/wiU51GlIhMdMzgFP6ljTP2/qr-code) look?

**Comment by locness3** [comment]
Have tyou made it by yourself ? BTW, please check you are conform with #171 .

**Comment by johnletey** [comment]
@locness3 I looked at the font-awesome design and then tried to use #171, although I'm not quite sure I did a good job ... but yes, I made it by myself.

**Comment by locness3** [comment]
> @locness3 I looked at the font-awesome design and then tried to use #171, although I'm not quite sure I did a good job ... but yes, I made it by myself.

Instead of doing what you're doing, use squares with 2px borders.

**Comment by johnletey** [comment]
@locness3 so no rounded edges ... but 2px borders?

**Comment by locness3** [comment]
> @locness3 so no rounded edges ... but 2px borders?

You should round edges with a 90degree angles (as states #171

**Comment by johnletey** [comment]
@locness3 I have updated the icon ... please take a look!

**Comment by locness3** [comment]
You should round the bottom-right thing, and make it a compound path.

**Comment by locness3** [comment]
More than just rounding them, you should make them strokes, no fills are allowed!

**Comment by johnletey** [comment]
@locness3 I have refactored with all of your changes ... unfortunately, it was really hard for me to get the two squares in the bottom-right thing to be apart of the rest of the vector for the bottom-right thing ... hopefully, this isn't a problem!

**Comment by locness3** [comment]
> @locness3 I have refactored with all of your changes ... unfortunately, it was really hard for me to get the two squares in the bottom-right thing to be apart of the rest of the vector for the bottom-right thing ... hopefully, this isn't a problem!

When you will have rounded the two squares at the bottom right, that will be perfect!

**Comment by johnletey** [comment]
@locness3 Not exactly quite sure how I'm supposed to round the square, as when I try to round them, they either change to nothing or a ring 😞 ... but, if I change the stroke width to 1px (which is clearly against #171), I get a perfectly rounded square!

**Comment by Aviortheking** [comment]
on Figma a single point with stroke isn't shown at all so don't worry about that

**Comment by locness3** [comment]
> @locness3 Not exactly quite sure how I'm supposed to round the square, as when I try to round them, they either change to nothing or a ring 😞 ... but, if I change the stroke width to 1px (which is clearly against #171), I get a perfectly rounded square!

Do a single path point with a stroke

**Comment by johnletey** [comment]
@locness3 How exactly do you do a "single path point with a stroke"?

**Comment by locness3** [comment]
> @locness3 How exactly do you do a "single path point with a stroke"?

Create one path point with your habitual pen tool I guess

**Comment by johnletey** [comment]
@locness3 Doesn't work ... when I try to do this, by creating a point with the pen tool, it doesn't show up even after I select different strokes

**Comment by locness3** [comment]
> @locness3 Doesn't work ... when I try to do this, by creating a point with the pen tool, it doesn't show up even after I select different strokes

Have you tried exporting it and oprening in a browser ?

**Comment by johnletey** [comment]
> Have you tried exporting it and oprening in a browser ?

@locness3 Just did ... not showing up

**Comment by locness3** [comment]
> > Have you tried exporting it and oprening in a browser ?
> 
> @locness3 Just did ... not showing up

🤔

**Comment by reinhart1010** [comment]
I'd like to suggest another design for the QR code icon:
![image](https://user-images.githubusercontent.com/17312341/74399555-c8691680-4e4d-11ea-8a32-e89a70b154a7.png)

This is heavily influenced by several Indonesian digital wallet apps that bear similar icons:
![image](https://user-images.githubusercontent.com/17312341/74399362-12053180-4e4d-11ea-9ffd-fd5f27217cf9.png) ![image](https://user-images.githubusercontent.com/17312341/74399371-17fb1280-4e4d-11ea-9169-c01c399586f6.png) ![image](https://user-images.githubusercontent.com/17312341/74399387-29441f00-4e4d-11ea-9791-59f261c1d91d.png)

I decided to use a rounded square on the bottom-right corner to be distinguishable from others, so people may still be able to interpret this as a normal QR code.

You can have a look of the SVG file at https://drive.google.com/open?id=14nO2J2PaODXfcgmB7ZIc3SXf4RPHzXQZ

**Comment by locness3** [comment]
Idk, what do you think @johnletey ?

**Comment by JelleBlaauw** [comment]
> @colebemis @locness3 @Aviortheking How does [this](https://www.figma.com/file/wiU51GlIhMdMzgFP6ljTP2/qr-code) look?

You've got the Font-Awesome flag in there, so no, I wouldn't do it like this.

**Comment by krrohitch** [comment]
Would like to make a suggestion here - [Figma link](https://www.figma.com/file/igAYKQcZytsnVGbMV0Nm7D/Feather-Icons-krrohitch?node-id=0%3A1)
![image](https://user-images.githubusercontent.com/22885068/88138675-c29d0b80-cc0b-11ea-950a-086ee506eadc.png)

**Comment by locness3** [comment]
@krrohitch To me, this is more "scan qr code" than "qr code".

**Comment by krrohitch** [comment]
You have a point there. I'll try one without the outer frame

**Comment by krrohitch** [comment]
Made a quick iteration. What do you think @locness3 ?
![image](https://user-images.githubusercontent.com/22885068/88146479-8c668880-cc19-11ea-9a61-082472e5e827.png)

**Comment by arjasco** [comment]
I attempted something similar using the grid icon as the basis.
<img width="74" alt="Screenshot 2020-08-01 at 11 34 51" src="https://user-images.githubusercontent.com/8780549/89099957-0bfb1100-d3eb-11ea-8745-28f011917adf.png">

Would love a native QR code icon that matches the rest of the icon set please!

**Comment by johnletey** [comment]
@arjasco Unfortunately, that icon has fills in it - which is against the Feather icon guidelines.

@krrohitch Well done 😍 Would love to see those icons add to @featherity 😉

**Comment by arjasco** [comment]
@johnletey Understood! It wasn't meant to be a candidate for this icon, just a play on an idea before coming to this issue.

How could we see the icon @krrohitch has produced make it in? This repo seems like is really isn't getting much attention which is a shame because it's a great icon set.

Does the mentioned fork contain all the current icons? I'm currently creating an SVG map with a subset of the icons I require, annoyingly could really do with a QR code one ☹️

**Comment by johnletey** [comment]
@arjasco Understand 😄

And yes, the fork is actively maintained by myself and the contributors. @ericfennis is working hard on publishing usable packages! (https://github.com/featherity/featherity/pull/52)

**Comment by arjasco** [comment]
@johnletey Looks great! Will keep an eye on the improvements and might look to swapping to this. Thanks!

**Comment by Qu4k3** [comment]
We are entering the third quarter of 2021, with a boom in the use of QRs and I see this thread with very interesting propositions stopped, any news about this icon request and its future incorporation? 👀

**Comment by CodeLake** [comment]
Been wanting to use the native feather QR-Code icon for a while and ending up using a work-around. Has the completed icon been published?

**Comment by reinhart1010** [comment]
Yeah, one has been merged to a fork of Feather icons - see lucide-icons/lucide#141

**Comment by CodeLake** [comment]
> Yeah, one has been merged to a fork of Feather icons - see [lucide-icons/lucide#141](https://github.com/lucide-icons/lucide/pull/141)

Thank you. I was wondering if there is a plan to add it to feather icon set natively?

**Comment by lauhon** [comment]
Bump, i would love this :)

**Comment by locness3** [comment]
This icon is available in @lucide-icons, a community fork of Feather : https://lucide.dev/icon/qr-code?search=qr

**Comment by lauhon** [comment]
Thanks for the information! :)

But wouldn't it be nice to merge it to the main repo?

For example, my usecase is to use the icons via [React Native Feather](https://www.npmjs.com/package/react-native-feather) which - as far as I know - only has access to the main icons and has no built in api to configure it to use the community fork. 

Im curious, whats the "right" way to go about this, should I make an effort to combine React Native Feather with the community icons, or should @feathericons include more community icons?

**Comment by karsa-mistmere** [comment]
[Lucide also has a standalone package for React Native](https://lucide.dev/docs/lucide-react-native), and has a whopping 975 icons compared to Feather's 287, I would advise you to just switch over to `lucide-react-native` from `react-native-feather`.

**Comment by lauhon** [comment]
Oh, nice. Thanks for pointing that out.

Almost makes me feel like a traitor, but in that case I'm gonna switch to lucide icons :D

**Comment by locness3** [comment]
> But wouldn't it be nice to merge it to the main repo?

Yes it would be. But the repository [has no active maintainers](https://github.com/feathericons/feather/issues/893).
